common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2StrategyParameters.java

```diff
@@ -0,0 +1,45 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.providers.oauth2;
+
+import android.content.Context;
+
+import androidx.annotation.Nullable;
+
+/**
+ * Abstract class for defining options relative to the construction of an {@link OAuth2Strategy}.
+ */
+public class OAuth2StrategyParameters {
```

Comment:
Do we really need this class with just one field?

Reply:
Not really, in fact, I don't want this parameter here **at all** 😅 

BUT, given that I need a `Context` to support asymmetric key generation inside the strategy my rationale for wrapping it up inside another class is that it provides an extensibility point for future params we may need to pass into the strategy.

For that reason, I'll leave as-is for now. We can remove later if no other future params come....